From 4e801fba6c9ff1c2740493c74396fb4a6c2cfd05 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Sat, 2 May 2026 01:04:29 -0700
Subject: [PATCH 21/21] rocm-libraries stinkytofu gcc 16/fedora 44 fixes

Stinkytofu broke the hibBLASLt build on fedora/gcc 16
due to missing definitions because c++ headers shipped
with newer gcc versions have more fine detailed control
for include files to be included.

Fixes following type of errors by including
ctdint.h or cstdint and climits when needed:

  error: unknown type name 'uint32_t'
and
  error: use of undeclared identifier 'UINT_MAX'

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 ...-16-fedora-44-uint32_t-include-fixes.patch | 91 +++++++++++++++++++
 1 file changed, 91 insertions(+)
 create mode 100644 patches/amd-mainline/rocm-libraries/0004-stinkytofu-gcc-16-fedora-44-uint32_t-include-fixes.patch

diff --git a/patches/amd-mainline/rocm-libraries/0004-stinkytofu-gcc-16-fedora-44-uint32_t-include-fixes.patch b/patches/amd-mainline/rocm-libraries/0004-stinkytofu-gcc-16-fedora-44-uint32_t-include-fixes.patch
new file mode 100644
index 0000000..49d910d
--- /dev/null
+++ b/patches/amd-mainline/rocm-libraries/0004-stinkytofu-gcc-16-fedora-44-uint32_t-include-fixes.patch
@@ -0,0 +1,91 @@
+From 8ef9eaeee472babebd8f383c091feac0f0fa2b6c Mon Sep 17 00:00:00 2001
+From: Mika Laitio <mika.laitio@amd.com>
+Date: Wed, 13 May 2026 13:43:41 -0700
+Subject: [PATCH 4/4] stinkytofu gcc 16/fedora 44 uint32_t include fixes
+
+Stinkytofu broke the hibBLASLt build on fedora/gcc 16
+due to missing definitions because c++ headers shipped
+with newer gcc versions have more fine detailed control
+for include files to be included.
+
+Fixes following type of errors by including
+ctdint.h or cstdint and climits when needed:
+
+  error: unknown type name 'uint32_t'
+and
+  error: use of undeclared identifier 'UINT_MAX'
+
+Signed-off-by: Mika Laitio <mika.laitio@amd.com>
+---
+ .../include/stinkytofu/serialization/logical/IRSerializer.hpp   | 1 +
+ shared/stinkytofu/src/serialization/asm/IRParser.cpp            | 1 +
+ shared/stinkytofu/src/serialization/asm/PatternParser.cpp       | 2 ++
+ shared/stinkytofu/src/transforms/asm/dag/CDNA5.hpp              | 1 +
+ shared/stinkytofu/src/transforms/asm/dag/ReadyQueue.hpp         | 2 ++
+ 5 files changed, 7 insertions(+)
+
+diff --git a/shared/stinkytofu/include/stinkytofu/serialization/logical/IRSerializer.hpp b/shared/stinkytofu/include/stinkytofu/serialization/logical/IRSerializer.hpp
+index e45f7162..5b58b5b5 100644
+--- a/shared/stinkytofu/include/stinkytofu/serialization/logical/IRSerializer.hpp
++++ b/shared/stinkytofu/include/stinkytofu/serialization/logical/IRSerializer.hpp
+@@ -23,6 +23,7 @@
+ 
+ #pragma once
+ 
++#include <cstdint>
+ #include <iostream>
+ #include <string>
+ #include <vector>
+diff --git a/shared/stinkytofu/src/serialization/asm/IRParser.cpp b/shared/stinkytofu/src/serialization/asm/IRParser.cpp
+index cbec4c5d..990943b7 100644
+--- a/shared/stinkytofu/src/serialization/asm/IRParser.cpp
++++ b/shared/stinkytofu/src/serialization/asm/IRParser.cpp
+@@ -25,6 +25,7 @@
+ 
+ #include <algorithm>
+ #include <cctype>
++#include <climits>
+ #include <iostream>
+ #include <optional>
+ #include <string_view>
+diff --git a/shared/stinkytofu/src/serialization/asm/PatternParser.cpp b/shared/stinkytofu/src/serialization/asm/PatternParser.cpp
+index 6846727a..a19bc393 100644
+--- a/shared/stinkytofu/src/serialization/asm/PatternParser.cpp
++++ b/shared/stinkytofu/src/serialization/asm/PatternParser.cpp
+@@ -22,6 +22,8 @@
+  * ************************************************************************ */
+ 
+ #include "stinkytofu/serialization/asm/PatternParser.hpp"
++#include <climits>
++#include <cstdint>
+ 
+ #include <bit>
+ #include <fstream>
+diff --git a/shared/stinkytofu/src/transforms/asm/dag/CDNA5.hpp b/shared/stinkytofu/src/transforms/asm/dag/CDNA5.hpp
+index 79a7fe6f..f66cdfdc 100644
+--- a/shared/stinkytofu/src/transforms/asm/dag/CDNA5.hpp
++++ b/shared/stinkytofu/src/transforms/asm/dag/CDNA5.hpp
+@@ -32,6 +32,7 @@
+ //
+ #include <algorithm>
+ #include <cassert>
++#include <cstdint>
+ #include <map>
+ 
+ #include "ReadyQueue.hpp"
+diff --git a/shared/stinkytofu/src/transforms/asm/dag/ReadyQueue.hpp b/shared/stinkytofu/src/transforms/asm/dag/ReadyQueue.hpp
+index d2ab4640..c3f0025c 100644
+--- a/shared/stinkytofu/src/transforms/asm/dag/ReadyQueue.hpp
++++ b/shared/stinkytofu/src/transforms/asm/dag/ReadyQueue.hpp
+@@ -22,6 +22,8 @@
+  * ************************************************************************ */
+ #pragma once
+ 
++#include <cstdint>
++#include <climits>
+ #include <cmath>
+ #include <iostream>  // TODO: don't use iostream.
+ #include <map>
+-- 
+2.52.0
+
-- 
2.52.0

